e2e/connect.test.tsx

```diff
@@ -5,49 +5,60 @@
 import '@ethersproject/providers'
 import 'jest-environment-hardhat'
 
-import { act, render, RenderResult, waitFor } from '@testing-library/react'
+import { render, RenderResult, waitFor } from '@testing-library/react'
 import { tokens } from '@uniswap/default-token-list'
 
 import { SwapWidget } from '../src'
 
 describe('connect', () => {
   let component: RenderResult
-  let wallet: HTMLElement
+  let account: HTMLElement
+  let connectWallet: HTMLElement
   let toolbar: HTMLElement
+  let tokenSelect: HTMLElement
 
   beforeEach(async () => {
     component = render(<SwapWidget tokenList={tokens} />)
-    wallet = await component.findByTestId('wallet')
+    connectWallet = await component.findByTestId('connect-wallet')
     toolbar = await component.findByTestId('toolbar')
+    tokenSelect = (await component.findAllByTestId('token-select'))[0]
   })
 
-  it('prompts for wallet connection in the Toolbar', async () => {
-    expect(wallet.hidden).toBeTruthy()
-    expect(toolbar.textContent).toBe('Connect wallet to swap')
+  describe('with no params, using fallback JSON RPC URL', () => {
+    it('prompts for wallet connection in the Toolbar and Wallet', async () => {
+      expect(toolbar.textContent).toBe('Connecting…')
+
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
+      expect(toolbar.textContent).toBe('Connect wallet to swap')
+      expect(tokenSelect.getAttribute('disabled')).toBeFalsy()
+    })
   })
 
-  describe('with jsonRpcUrlMap', () => {
-    it('prompts for wallet connection in the Wallet', async () => {
-      act(() => component.rerender(<SwapWidget tokenList={tokens} jsonRpcUrlMap={{ 1: [hardhat.url] }} />))
-      expect(wallet.hidden).toBeTruthy()
+  describe('with integrator jsonRpcUrlMap', () => {
+    it('prompts for wallet connection in the Wallet and Toolbar', async () => {
```

Comment:
I think it makes more sense to test at once? - it's one test case (passing in `jsonRpcUrlMap={{ 1: [hardhat.url] }}`), and we're just checking that the textContent in the wallet chip + toolbar appear as expected

Reply:
when tests fail it's better to know *exactly* what failed, instead of needing to go fiddle with the test and double check the logs to see which thing that was being tested failed